addons/website_sale/__init__.py

```diff
@@ -17,3 +17,19 @@ def uninstall_hook(cr, registry):
     multi_company_rules = pl_rule or env['ir.rule']
     multi_company_rules += pl_item_rule or env['ir.rule']
     multi_company_rules.write({'active': True})
+
+
+def post_init_hook(cr, registry):
+    ''' Need to add product filters to previously created website
+    '''
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    action = env.ref('website_sale.dynamic_snippet_products_action', raise_if_not_found=False)
+    if action:
+        for website in env['website'].search([]):
+            env['website.snippet.filter'].create({
+                'action_server_id': action.id,
+                'field_names': 'display_name,description_sale,image_512,list_price',
+                'limit': 16,
+                'name': _('Products'),
+                'website_id': website.id,
+            })
```

Comment:
Copy pasting code is generally not great. This should probably be encapsulated and reused. It would be a lot easier to figure when someone will read the code, as greping the new method name would return 2 places, the one here and the one in bootstrap_website.

This is especially the case when it is not mentioned, it makes the review harder than it should, here you should mention commit https://github.com/odoo/odoo/commit/01fafecf98aba5886c5fe836e2a81f2551155f35

Reply:
well, since most of the code here is just the parameters of the create method, I was thinking it doesn't make much since to encapsulate it as it is somehow already encapsulated in the create method?

All the rest is just logic and parameters, encapsulating the logic is also not useful because it is a bit different from elsewhere (due to searching all websites and checking the action first)

What do you think?
also do you mean mention the commit in a comment in the code or in the commit message?

Reply:
I would have still encapsulated, but I got your point.
If I were to keep it like you did, I would at least add a comment here
```py
# Do the same as in _bootstrap_snippet_filters()
```
And same on `_bootstrap_snippet_filters()`, add 
```py
# The same behavior is done in the post_init hook
```

That really is just to ease code reading later, if you agree.

> also do you mean mention the commit in a comment in the code or in the commit message?

Commit msg :)